controllers/front/validation.php

```diff
@@ -98,122 +98,135 @@ public function postProcess()
-                __PS_BASE_URI__ .
-                'index.php?controller=order-confirmation&id_cart=' .
-                (int) $cart->id .
-                '&id_module=' .
-                (int) $blockonomics->id .
-                '&id_order=' .
-                $blockonomics->currentOrder .
-                '&key=' .
+            $mes = "Adr BTC : " . $address;
+            $blockonomics->validateOrder(
+                (int) $cart->id,
+                Configuration::get('BLOCKONOMICS_ORDER_STATE_WAIT'),
+                $total,
+                $blockonomics->displayName,
+                $mes,
+                $mailVars,
+                (int) $currency->id,
+                false,
                 $customer->secure_key
-        );
+            );
 
-        $mes = "Adr BTC : " . $new_address;
-        $blockonomics->validateOrder(
-            (int) $cart->id,
-            Configuration::get('BLOCKONOMICS_ORDER_STATE_WAIT'),
-            $total,
-            $blockonomics->displayName,
-            $mes,
-            $mailVars,
-            (int) $currency->id,
-            false,
-            $customer->secure_key
-        );
+            $id_order = $blockonomics->currentOrder;
+            // Add the backup cart to user
+            $new_cart->id_customer = $old_cart_customer_id;
+            $new_cart->save();
+            if ($new_cart->id) {
+                $this->context->cookie->id_cart = (int) $new_cart->id;
+                $this->context->cookie->write();
+            }
+            $id_cart = (int) $new_cart->id;
+            $new_cart->secure_key = $old_cart_secure_key;
 
-        // Add the backup cart to user
-        $new_cart->id_customer = $old_cart_customer_id;
-        $new_cart->save();
-        if ($new_cart->id) {
-            $this->context->cookie->id_cart = (int) $new_cart->id;
-            $this->context->cookie->write();
+            Db::getInstance()->Execute(
+                "INSERT INTO " .
+                    _DB_PREFIX_ .
+                    "blockonomics_bitcoin_orders (id_order, id_cart, timestamp,  ".
+                    "addr, txid, status,value, bits, bits_payed) VALUES
+                    ('" .
+                    (int) $blockonomics->currentOrder .
+                    "','" .
+                    (int) $id_cart .
+                    "','" .
+                    (int) $current_time .
+                    "','" .
+                    pSQL($address) .
+                    "', '', -1,'" .
+                    (float) $total .
+                    "','" .
+                    (int) $bits .
+                    "', 0)"
+            );
+        } else {
+            $address = $order['addr'];
+            $id_order = $order['id_order'];
+            $current_time = $order['timestamp'];
+            $time_remaining = $this->get_time_remaining($order);
+            //if value of cart has changed because the user has added or removed products from the cart
```

Comment:
This comment is incorrect and can be moved/removed.

Reply:
True, the comment has now been removed.